src/main/java/io/dataspaceconnector/utils/RuleUtils.java

```diff
@@ -328,6 +331,63 @@ public static URI getAllowedPurpose(Rule rule) {
         Constraint constraint = (Constraint)rule.getConstraint().get(0);
         URI allowedPurpose = constraint.getRightOperandReference();
         return allowedPurpose;
-    }  
-    
+    }
+
+
+    //TECNALIA-ICT-OPTIMA: New function for new rule
+    /**
+     * Returns the allowed Security Restriction.
+     *
+     * @param rule the policy constraint object
+     * @return the URI of the Purpose
+     */
+    public static URI getAllowedSecurityRestriction(Rule rule) {
+        Constraint constraint = (Constraint)rule.getConstraint().get(0);
+        URI allowedSecurityRestriction = constraint.getRightOperandReference();
```

Comment:
Why getRightOperandReference() and not rightOperand()?

Based on your provided test, I've generated the example, and the value of allowedSecurityRestriction is always null.